app/forms/provider_interface/offer_wizard.rb

```diff
@@ -90,21 +92,27 @@ def save_and_go_to_next_step(step)
       next_step(step)
     end
 
+    def query_service
+      @query_service ||= GetChangeOfferOptions.new(
+        user: ProviderUser.find(provider_user_id),
+        current_course: ApplicationChoice.find(application_choice_id).offered_course,
+      )
+    end
+
     def available_study_modes
-      Course.find(course_id).available_study_modes_from_options
+      query_service.available_study_modes(course: Course.find(course_id))
     end
 
     def available_course_options
-      CourseOption.where(course_id: course_id, study_mode: study_mode)
+      query_service.available_course_options(course: Course.find(course_id), study_mode: study_mode)
```

Comment:
Similar to above, here we should be able to reuse a defined `course` reusable method.